swarm/src/dial_opts.rs

```diff
@@ -297,29 +297,36 @@ impl WithoutPeerIdWithAddress {
     }
 }
 
-/// The available conditions under which a new dialing attempt to
-/// a known peer is initiated.
-///
-/// ```
-/// # use libp2p_swarm::dial_opts::{DialOpts, PeerCondition};
-/// # use libp2p_identity::PeerId;
-/// #
-/// DialOpts::peer_id(PeerId::random())
-///    .condition(PeerCondition::Disconnected)
-///    .build();
-/// ```
-#[derive(Debug, Copy, Clone, Default)]
-pub enum PeerCondition {
-    /// A new dialing attempt is initiated _only if_ the peer is currently
-    /// considered disconnected, i.e. there is no established connection
-    /// and no ongoing dialing attempt.
-    #[default]
-    Disconnected,
-    /// A new dialing attempt is initiated _only if_ there is currently
-    /// no ongoing dialing attempt, i.e. the peer is either considered
-    /// disconnected or connected but without an ongoing dialing attempt.
-    NotDialing,
-    /// A new dialing attempt is always initiated, only subject to the
-    /// configured connection limits.
-    Always,
+bitflags::bitflags! {
+    /// The available conditions under which a new dialing attempt to
+    /// a known peer is initiated.
+    ///
+    /// ```
+    /// # use libp2p_swarm::dial_opts::{DialOpts, PeerCondition};
+    /// # use libp2p_identity::PeerId;
+    /// #
+    /// DialOpts::peer_id(PeerId::random())
+    ///    .condition(PeerCondition::Disconnected)
+    ///    .build();
+    /// ```
+    #[derive(Debug, Copy, Clone)]
+    pub struct PeerCondition: u32 {
+        /// A new dialing attempt is always initiated, only subject to the
+        /// configured connection limits.
+        const None = 0b0000;
+        /// A new dialing attempt is initiated _only if_ the peer is currently
+        /// considered disconnected, i.e. there is no established connection
+        /// and no ongoing dialing attempt.
+        const Disconnected = 0b0001;
+        /// A new dialing attempt is initiated _only if_ there is currently
+        /// no ongoing dialing attempt, i.e. the peer is either considered
+        /// disconnected or connected but without an ongoing dialing attempt.
+        const NotDialing = 0b0010;
```

Comment:
Are these docs still true? Now that we can combine them, having them say "only if" doesn't work, does it?